### PR TITLE
Mdc Dataset

### PR DESCRIPTION
**Pull request by EthanMarx**

### CODE REVIEW

**Comment by EthanMarx** [comment]
@deepchatterjeeligo take a look. 

(note I also added infra for using `asd` info. I have our dataloaders / augmentors automatically pass this along to the embedding, and let the embedding decide whether or not it wants to use it)

File: amplfi/train/data/datasets/testing/debug-1.ipynb
Comment (deepchatterjeeligo) [inline amplfi/train/data/datasets/testing/debug-1.ipynb:1]: Let's add `*ipynb` to gitignore

File: amplfi/train/data/datasets/testing/mdc.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+        # scale parameters
+        parameters = self.scale(parameters)
+
+        # calculate asds
+        freqs = torch.fft.rfftfreq(X.shape[-1], d=1 / self.hparams.sample_rate)
+        num_freqs = len(freqs)
+        psds = torch.nn.functional.interpolate(
+            psds, size=(num_freqs,), mode="linear"
+        )
```
Comment (deepchatterjeeligo) [inline amplfi/train/data/datasets/testing/mdc.py:116]: what is this for?

**Review by deepchatterjeeligo — COMMENTED** [review]
Looks good overall @EthanMarx. Have you made a diagnostic plots for the strain after whitening to ensure that things look good?

File: amplfi/train/data/datasets/testing/mdc.py (hunk 1 above)
Comment (EthanMarx) [reply amplfi/train/data/datasets/testing/mdc.py:116]: this makes sure the psds are the same length of the data so they can be concatenated before passing to the embedding. (This is done under the hood in the `Whitener`)

**Comment by EthanMarx** [comment]
Yeah I made qscans and strain plots and things looked fine. The signals were placed at the proper location.

File: amplfi/train/data/datasets/testing/mdc.py (hunk 1 above)
Comment (deepchatterjeeligo) [reply amplfi/train/data/datasets/testing/mdc.py:116]: Let's put that down as a one-line comment?

**Review by deepchatterjeeligo — APPROVED** [review]

**Pull request merged**